### ISSUE TITLE
Cannot start Job Card if related to Work Order not started yet

### ISSUE THREAD

**Issue by FHenry** [frappe/erpnext#29071]
Create a WO (from BOM with operation and Material transfer for WO)
Submit WO => Start WO => Stock transfer to WIP warehouse

Unfortunately you don't have enough stock to transfer to WIP, so you have WO in Submit status, Stock Entry in Draft, and Job card in draft.

On job card, you shouldn't be able to start counting time if work order is not started.

**Comment by sharmanakul** [comment]
hi @FHenry , thanks for creating the feature request. 
DO you think there could be operations in the work order which can be started without transferring material? For example: Operations X could simply be the preparation of the workstation. Operation Y could be something wherein the actual materials are consumed.

If you need to control the item transfer specific to the operation, then you can use 'Material transfer against' job card. Here also you can also start the job without first transferring the material but you won't be able to submit the job card. 

Let me know if you have some other use-case.

**Comment by FHenry** [comment]
hi @sharmanakul , thank's for the reply.
Yes, thank it's a good point.
I didn't think about that. 
So in this case I will prefers fer to add a new field into Bom's Operation Table to allow to start job card before Work  Order is started. 

But the first operation ask, by ERPNext, when a Work Order is started (material transfert on Work Order),  is the Stock Material Transfer (from A to WorkInProgress). The Stock Material Transfer stay in Draft, and Work Order is not started, if you do not have enough stock to transfer to production. 
So basically no operations could be done if you don't have enough stock to start it. It's the same with Material Transfer against Operation Card,.
If you said  the first operation is prepare the work station, why would you do it if you don't have enough stock to start to work ?
If you said the first operation is the physical stock transfer, you should not send your carrier if you don't have enough to pick ?
In what use case you can start operation without required materials to work on it ?

**Comment by FHenry** [comment]
hi @sharmanakul, i've update the PR to take care about "no WIP transfert needed"
Let me know if this feature is interesting for Frappe.inc, if not I can move over and focus on other ERPNext matter

**Comment by sharmanakul** [comment]
@ankush can you check the linked PR?
